Themes/Christmas_Season_2/index.template.php

```diff
@@ -107,7 +107,7 @@ function template_main_above()
 	// If we're viewing a topic, these should be the previous and next pages of the same topic, respectively.
 	// For some reason SMF developers thought that the best thing to put down as the "next page" is the next thread.
 	// That's retarded, and thus now fixed.
-	if (!empty($context['current_topic']))
+	if (!empty($context['current_topic']) && !empty($context["page_index"]))
 	{
 		$next_page = $context['start'] + $context['messages_per_page'];
 		$prev_page = $context['start'] - $context['messages_per_page'];
```

Comment:
The actual errors generated in the log are:

>8: Undefined index: start
>8: Undefined index: messages_per_page

Why not test the keys we actually need to use in this if block?

```suggestion
	if (!empty($context['current_topic']) && !empty($context["start"]) && !empty($context["messages_per_page"]))
	{
		$next_page = $context['start'] + $context['messages_per_page'];
		$prev_page = $context['start'] - $context['messages_per_page'];
```

Reply:
My logic was to try and find a test that would confirm we're actually viewing a thread, on the off chance that `$context[start]` and `$context[messages_per_page]`  may exist in other contexts (now or in the future). I believe the current test achieves this, and the existence of the two keys should be guaranteed at that point.

Your approach is definitely easier to read/understand, and *probably* functionally equivalent. I have no strong preference either way - feel free to make the call.

Reply:
I've made the change I suggested, along with a check for `total_visible_posts` (which is also used in these if blocks). It just seems more correct to me to test for what's in use, although I agree it should be functionally equivalent.

If you can review/approve my latest commit, we can merge and deploy this today.